cmd/state-svc/internal/resolver/resolver.go

```diff
@@ -88,6 +90,8 @@ func (r *Resolver) Close() error {
 func (r *Resolver) Query() genserver.QueryResolver { return r }
 
 func (r *Resolver) Version(ctx context.Context) (*graph.Version, error) {
+	defer handlePanics(recover(), debug.Stack())
```

Comment:
There's a good chance that this will not work as expected. The context in which `recover` and `debug.Stack` are called is the moment that the defer is called. This means that it should be `nil`. Please consider: https://go.dev/play/p/dfuoRApHgeC - The panic value is not handled and halts the program abruptly.

Please also consider: https://go.dev/play/p/nic95m6QXbT - This shows that an anonymous function permits the function being checked within the defer statement to have the expected context of "at the end of the function".